### PR TITLE
Final changes in credential cache function

### PR DESCRIPTION
**Pull request by roshani-delinea-github**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py [hunk 1]
```diff
@@ -478,8 +478,28 @@ def secret_rpc_changepassword_command(client, secret_id: str = '', newpassword:
...
 
 
+def get_credentials(client,secret_id):
+    obj={}
+    secret = client.getSecret(secret_id)
+    items = secret.get('items')
+    for item in items:
+        if item.get('fieldName') == 'Username':
+            username = item.get('itemValue')
+        if item.get('fieldName') == 'Password':
```
Comment (yaakovpraisler) [inline Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py:488]: Why does the field `fieldName` in item will be both (Username and Password)? Should the second one (for password) be `itemValue` as in line 489?

File: Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py [hunk 2]
```diff
@@ -478,8 +478,28 @@ def secret_rpc_changepassword_command(client, secret_id: str = '', newpassword:
...
+            password = item.get('itemValue')
+            obj = {
+                "user": username,
+                "password": password,
+                "name": str(secret.get('id'))
+            }
```
Comment (yaakovpraisler) [inline Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py:494]: Why does `obj` configured only if item is password? i.e., the obj definition is intended under `if item.get('fieldName') == 'Password'` so user_name will never have a value in obj.

Did you mean this?
```suggestion
        if item.get('fieldName') == 'Password':
            password = item.get('itemValue')
        obj = {
            "user": username,
            "password": password,
            "name": str(secret.get('id'))
        }

```

**Review by yaakovpraisler — CHANGES_REQUESTED** [review]
Hi @roshani-delinea-github, Thank you for your contribution!
Overall looks good, just a small thing that I may not have understood correctly -

Also please bump the version of the pack in the pack_metadata.json and write release notes for this version (you can se [here](https://xsoar.pan.dev/docs/contributing/checklist) for more details we need to take care when contributing).

Thanks again :)

File: Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py (hunk 1 above)
Comment (roshani-delinea-github) [reply Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py:488]: It is to extract value from the dictionary item.

File: Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py (hunk 2 above)
Comment (roshani-delinea-github) [reply Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py:494]: We want to save all the data in obj. I move the obj as you suggested it shows the error for the username is referenced before the assignment. Current implementation is working fine

File: Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py (hunk 1 above)
Comment (yaakovpraisler) [reply Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py:490]: Yeah I got that, but you're extracting the field `fieldName` twice, in line 488 for `Username` and in line 490 for `Password`

**Review by yaakovpraisler — COMMENTED** [review]
Hi @roshani-delinea-github,
You need also bump the pack version and add [release notes](https://xsoar.pan.dev/docs/documentation/release-notes)

Please feel free to reach out to me with any questions or issues - I'm available here or on slack :)
Thanks again

File: Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py (hunk 1 above)
Comment (roshani-delinea-github) [reply Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py:490]: Yes. The fieldname is used to store the name of the field example username and password and their value is stored in itemvalue field. I want to extract both values. We have input like below : 
'items': [
            {
                'itemId': 14,
                'fileAttachmentId': None,
                'filename': None,
                'itemValue': 'user',
                'fieldId': 111,
                'fieldName': 'Username',
                'slug': 'username',
                'fieldDescription': 'The Unix Machine Username.',
                'isFile': False,
                'isNotes': False,
                'isPassword': False
            },
            {
                'itemId': 15,
                'fileAttachmentId': None,
                'filename': None,
                'itemValue': 'password',
                'fieldId': 110,
                'fieldName': 'Password',
                'slug': 'password',
                'fieldDescription': 'The password of the Unix Machine.',
                'isFile': False,
                'isNotes': False,
                'isPassword': True
            }
        ]

**Comment by yaakovpraisler** [comment]
Closing as duplicate. Will continue in [this PR](https://github.com/demisto/content/pull/24467)

**Pull request closed without merging**